cjwkernel/types.py

```diff
@@ -634,10 +644,10 @@ def trans(
 
     @classmethod
     def from_dict(cls, value: Dict[str, Any]) -> I18nMessage:
-        return cls(value["id"], value["arguments"])
+        return cls(value["id"], value["arguments"], value["source"])
```

Comment:
This isn't backwards-compatible. We already have thousands of I18nMessages in our database, and none of them have a `source`. Use `value.get("source", None)`.